src/modules/infobox/components/FeatureProfile.vue

```diff
@@ -218,27 +230,35 @@ export default {
             }
         },
         async createProfileChart() {
-            const data = await this.getProfile(this.featureCoordinates)
+            const data = await this.getProfile()
             this.profileChart.create(data)
             const areaChartPath = this.profileChart.group.select('.profile-area')
             const glass = this.profileChart.glass
             this.attachPathListeners(areaChartPath, glass)
             return this.profileChart.element
         },
         async updateProfileChart(size) {
-            const data = await this.getProfile(this.featureCoordinates)
+            const data = await this.getProfile()
             this.profileChart.update(data, size)
             return this.profileChart.element
         },
-        async getProfile(coordinates) {
+        async getProfile(apiFunction = profileJson, coordinates = this.featureCoordinates) {
```

Comment:
it's declared async, but there's no `await`, the promise is returned directly.
As it's the same method as the one called by `onCSVDownload`, you could `await` here the API call, and `try...catch` and log any error